### ISSUE TITLE
NB-MLM: Efficient Domain Adaptation of Masked Language Models for Sentiment Analysis

### ISSUE THREAD

**Issue by Vannora9** [Vannora9/Sentiment_Analysis#1]
## 0．論文詳細
【Authors】Nikolay Arefyev et.al
【CPublisher】EMNLP
【Submission】2021
【URL】https://aclanthology.org/2021.emnlp-main.717.pdf

【Abstract】
    While Masked Language Models (MLM) are pre-trained on massive datasets, the additional training with the MLM objective on domain or task-specific data before fine-tuning for the final task is known to improve the final performance. This is usually referred to as the domain or task adaptation step. However, unlike the initial pre-training, this step is performed for each domain or task individually and is still rather slow, requiring several GPU days compared to several GPU hours required for the final task fine-tuning. **_We argue that the standard MLM objective leads to inefficiency when it is used for the adaptation step because it mostly learns to predict the most frequent words, which are not necessarily related to a final task._** We propose a technique for more efficient adaptation than _focuses on predicting words with large weights of the Naive Bayes classifier trained for the task at hand, which are likely more relevant than the most frequent words._ The proposed method provides faster adaptation and better final performance for sentiment analysis compared to the standard approach.

虽然屏蔽语言模型（MLM）是在大量的数据集上进行预训练的，但在为最终任务进行微调之前，用MLM目标在特定领域或任务的数据上进行额外的训练，已知可以提高最终的性能。这通常被称为领域或任务适应步骤。然而，与最初的预训练不同，这个步骤是为每个领域或任务单独进行的，而且仍然相当缓慢，需要几个GPU天，而最终任务微调需要几个GPU小时。我们认为，**_标准的MLM目标在用于适应步骤时导致效率低下，因为它主要学习预测最频繁的词，而这些词不一定与最终的任务有关_**。我们提出了一种更有效的适应技术，其重点是_预测具有大权重的Naive Bayes分类器为手头的任务所训练的词，这些词可能比最频繁的词更相关_。与标准方法相比，所提出的方法为情感分析提供了更快的适应和更好的最终性能。

## 1．Research summary
提出一种更有效的领域和任务适应技术，在特定数据集上训练，提高性能的同时缩短了训练时间。
Proposing a more effective domain and task method to train on the specific dataset, the result not only improves the performance but also shortens the training time.

## 2．Problem setting and solutions
Standard MLM model 需要花费大量时间学习和存储出现次数最多的单词，但是出现频率最多不一定与最后的task相关。
所以本文提出一种有效的领域和任务适应技术，通过选择训练与final task相关的单词，从而提高最终性能。
The Standard MLM model spends a lot of time learning and storing the most frequent words, but the most frequent occurrences are not necessarily related to the last task.
So this paper proposes an effective domain and task adaptation technique to improve the final performance by selecting and training words relevant to the final task.

## 3．The key to technology and method
相比于uniform MLM model 选择15%进行loss计算， NB-MLM model选择从非均匀分布中提取 15% 的位置，从而为包含具有高特征重要性的子词的位置提供更高的概率。对于二分类任务，通过下面的Bayes获得权重。
![image](https://user-images.githubusercontent.com/86482592/180591691-37506912-54a1-43d3-a487-d927bb69b67a.png)

## 4． Validity verification of claims
1. 对于domain adaptation， 使用去除重复行的Amazon Reviews dataset， 训练集21M, 验证集10k。
2. 对于task adaptation， 直接使用Gururangan et al. (2020) to make the results of our experiments directly comparable with their results.

未进行权重设置以及设置权重后target tokens的对比：
![image](https://user-images.githubusercontent.com/86482592/180591917-0fe89a75-a355-4cbd-8521-3e09af73eed3.png)

standard MLM model和NB-MLM model对于domain 和 task的对比：
![image](https://user-images.githubusercontent.com/86482592/180591957-0c3d3ea2-0eef-4b49-9264-218f593af722.png)

对比各种模型和数据集的结果如下：
![image](https://user-images.githubusercontent.com/86482592/180591895-d96b3119-5697-4893-a330-de0f48d7796c.png)

## 5．Points to discuss
Why not use stop words for the standard MLM?
